File: chap5/Quotient.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let rec quotient dividende diviseur =
   if dividende < diviseur then
     0
   else
     1 + quotient (dividende - diviseur) diviseur

let _ =
  print_int ( quotient 17 3 );
  print_newline ()

File: chap5/Quotient.java
// Ces programmes sont sous licence CeCILL-B V1.
class Quotient {

  static int quotient (int dividende, int diviseur) {
    if (dividende < diviseur) {
      return 0;}
    else {
	return (1 + quotient(dividende - diviseur, diviseur));}}

  public static void main (String [] args) {
    System.out.println(quotient(17,3));}}
